python/python/lance/torch/kmeans.py

```diff
@@ -0,0 +1,140 @@
+#  Copyright (c) 2023. Lance Developers
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from typing import Optional, Union
+
+import numpy as np
+import pyarrow as pa
+import torch
```

Comment:
just double checking - torch is an optional dependency right? Can we make this a safe import that if the user hasn't installed torch explicitly this would actually output some user friendly message?

Reply:
The pytorch installation check is done here

https://github.com/lancedb/lance/blob/5aad966c43b0b96ce9f7ae64b1b29aa245026f50/python/python/lance/torch/__init__.py#L17-L20